File: python scripts/insertion_sort.py
def insert(list, rightIndex, value):
    while rightIndex >=0 and list[rightIndex] > value:
        list[rightIndex + 1] = list[rightIndex]
        rightIndex -= 1
    list[rightIndex + 1] = value
        
def insertion_sort(list):
    for i in range(1, len(list)):
        insert(list, i-1, list[i])

list = [22, 11, 99, 88, 9, 7, 42]
insertion_sort(list)
print("Sorted list", str(list))

"""
0,11 > 22,22,99,88,9,7,42 > 11,22,99,88,9,7,42
1,99 > 11,22,99,88,9,7,42
2,88 > 11,22,99,99,9,7,42 > 11,22,88,99,9,7,42
3,9 > 11,22,88,99,99,7,42 > 11,22,88,88,99,7,42 > 11,22,22,88,99,7,42 > 11,11,22,88,99,7,42 > 9,11,22,88,99,7,42
4,7 > 9,11,22,88,99,99,42 > 9,11,22,88,88,99,42 > 9,11,22,22,88,99,42 > 9,11,11,22,88,99,42 > 9,9,11,22,88,99,42 > 7,9,11,22,88,99,42
5,42 > 7,9,11,22,88,99,99 > 7,9,11,22,88,88,99 > 7,9,11,22,42,88,99
"""

File: javascripts/insertion_sort.js
var insert = function(array, rightIndex, value){
    for(var j=rightIndex; j>=0 && array[j] > value; j--){
        array[j+1] = array[j];
    }
    array[j+1] = value;
}

var insertionSort = function(array){
    for(var i = 1; i<array.length; i++){
        insert(array,i-1,array[i]);
    }
};
var array = [22, 11, 99, 88, 9, 7, 42];
insertionSort(array);
console.log(array);


/*var array = [3, 5, 7, 11, 13, 2, 9, 6];

insert(array, 4, 2);
console.log("Array after inserting 2:  " + array);
Program.assertEqual(array, [2, 3, 5, 7, 11, 13, 9, 6]);

insert(array, 5, 9);
console.log("Array after inserting 9:  " + array);
Program.assertEqual(array, [2, 3, 5, 7, 9, 11, 13, 6]);

insert(array, 6, 6);
console.log("Array after inserting 6:  " + array);
Program.assertEqual(array, [2, 3, 5, 6, 7, 9, 11, 13]);*/
